account_payment_term_extension/models/account_payment_term_line.py

```diff
@@ -0,0 +1,159 @@
+        },
+    )
+    value = fields.Selection(
+        selection_add=[
+            ("percent_amount_untaxed", "Percent (Untaxed amount)"),
+            ("fixed",),
+        ],
+        ondelete={"percent_amount_untaxed": lambda r: r.write({"value": "percent"})},
+    )
+
+    payment_days = fields.Char(
+        string="Payment day(s)",
+        help="Put here the day or days when the partner makes the payment. "
+        "Separate each possible payment day with dashes (-), commas (,) "
+        "or spaces ( ).",
+    )
+
+    def _get_delay_type(self):
+        delay_setting = self.env.company.payment_terms_delay_type
+        payment_terms_delay_type = delay_setting or False
+
+        delay_type = [
+            ("days_after", "Days after invoice date"),
+            ("days_after_end_of_month", "Days after end of month"),
+            ("days_after_end_of_next_month", "Days after end of next month"),
+            ("days_end_of_month_on_the", "Days end of month on the"),
+        ]
+
+        weeks_delay_type = [
+            ("weeks_after", "Weeks after invoice date"),
+            ("weeks_after_end_of_month", "Weeks after end of month"),
+            ("weeks_after_end_of_next_month", "Weeks after end of next month"),
+        ]
+        months_delay_type = [
+            ("months_after", "Months after invoice date"),
+            ("months_after_end_of_month", "Months after end of month"),
+        ]
+
+        if payment_terms_delay_type:
+            if payment_terms_delay_type:
+                if payment_terms_delay_type == "months":
+                    delay_type += months_delay_type
+                elif payment_terms_delay_type == "weeks":
+                    delay_type += weeks_delay_type
+                elif payment_terms_delay_type == "weeks_and_months":
+                    delay_type += weeks_delay_type + months_delay_type
+        return delay_type
+
+    def _decode_payment_days(self, days_char):
+        # Admit space, dash and comma as separators
+        days_char = days_char.replace(" ", "-").replace(",", "-")
+        days_char = [x.strip() for x in days_char.split("-") if x]
+        days = [int(x) for x in days_char]
+        days.sort()
+        return days
+
+    @api.constrains("payment_days")
+    def _check_payment_days(self):
+        for record in self:
+            if not record.payment_days:
+                continue
+            try:
+                payment_days = record._decode_payment_days(record.payment_days)
+                error = any(day <= 0 or day > 31 for day in payment_days)
+            except Exception:
+                error = True
+            if error:
+                raise exceptions.UserError(
+                    self.env._("Payment days field format is not valid.")
+                )
+
+    def _get_due_date(self, date_ref):
+        res = super()._get_due_date(date_ref)
+        due_date = fields.Date.from_string(date_ref) or fields.Date.today()
```

Comment:
Please use `context_today` to prevent issues in countries where `UTC` is negative.
Additionally, it is not necessary to cast `from_string` because now it is the native `date type`.

```suggestion
        due_date = date_ref or fields.Date.context_today(self)
```

Reply:
I'm a afraid I'm rejecting this one.  Not because I don't agree, it's just they are not originating here. They come from Odoo CE, i.e. `super()`.

Technically speaking `super()` dictates the signature of this method, and `super()` applies `from_string()` to `date_ref`. See: https://github.com/odoo/odoo/blob/d9d1c4844b81cd9be993d0557a2d5a9726ac5739/addons/account/models/account_payment_term.py#L305

You can't rule out in principle some module exists out there that is relying on `date_ref` possibly being a string, or that the default is `Date.today()` and not `Date.context_today()`.

Rght or wrong that's the contract that method offers in Odoo CE, we should not break it unless we have an extremely good reasons to do so.

Reply:
I understand your position. In fact, I noticed that Odoo has obsolete code, but this does not imply that we cannot improve it.

But anyway, if this is working, then it's OK for me. Thanks!